src/main/java/com/savvato/tribeapp/services/AttributesServiceImpl.java

```diff
@@ -75,4 +85,26 @@ public boolean isGivenAdverbFound(String adverb) {
     public boolean isGivenPrepositionFound(String preposition) {
         return this.prepositionRepository.findByWord(preposition).isPresent();
     }
+
+    @Override
+    public Optional<List<AttributeDTO>> getAttributesByUserId(Long userId) {
+
+        List<AttributeDTO> attributes = new ArrayList<>();
+
+        // Get all user phrases as phraseDTOs
+        Optional<List<PhraseDTO>> optUserPhraseDTOs = phraseService.getListOfPhraseDTOByUserId(userId);
+
+        // If there are phrases, build DTO and add to attributes list
+        if(optUserPhraseDTOs.isPresent()) {
+            List<PhraseDTO> attribute = optUserPhraseDTOs.get();
+            for(PhraseDTO phrase : attribute) {
+                AttributeDTO attributeDTO = AttributeDTO.builder().build();
+                attributeDTO.phrase = phrase;
```

Comment:
I think you can do `AttributeDTO.builder().phrase(phrase).build()`, basically combining lines 101 and 102. I'm 87% sure that will work. Let me know.

Reply:
That worked! Learned something new. Thanks!
reference to confirm: https://howtodoinjava.com/lombok/lombok-builder-annotation/